core/src/test/java/org/web3j/protocol/core/filters/FilterTester.java

```diff
@@ -69,7 +74,7 @@ public void setUp() {
         when(web3jService.send(any(Request.class), eq(EthFilter.class)))
                 .thenReturn(ethFilter);
         when(web3jService.send(any(Request.class), eq(EthLog.class)))
-                .thenReturn(ethLog);
+            .thenReturn(ethLog).thenReturn(notFoundFilter).thenReturn(ethLog);
         when(web3jService.send(any(Request.class), eq(EthUninstallFilter.class)))
```

Comment:
Is this getting called, given only one request is being sent?

Reply:
The first request is always to create a filter, and the subsequent requests for an update on it. So when we create filter normally then we imitate filtr not found query and filter will be resubscribe automatically, so yes test will work if there only one update request.

Reply:
👍